impl/src/main/java/org/apache/myfaces/context/servlet/PartialViewContextImpl.java

```diff
@@ -579,6 +579,11 @@ public VisitResult visit(VisitContext context, UIComponent target)
             {
                 ((EditableValueHolder)target).resetValue();
             }
+            else if (!VisitContext.ALL_IDS.equals(context.getIdsToVisit()))
+            {
+                // Render ID didn't specifically point an EditableValueHolder. Visit all children as well.
+                target.visitTree(VisitContext.createVisitContext(FacesContext.getCurrentInstance(), null, context.getHints()), this);
```

Comment:
can we make the visitContext static or reuseable?
currently this would create tons of the same context object, not quite good from performance perspective
just look at PF one

Reply:
and please create a unit test for it

Reply:
i saw that about the FacesContext but was trying not to make the change too invasive. I will rework this today.

Reply:
yeah sure
ping if you need help

Reply:
OK @tandraschko i might need a hint on this one.  PF does his slighlty differnetly by just calling visitTree where in this case BalusC specifically creates a new VisitContext?